sds_data_manager/lambda_code/SDSCode/api_lambdas/utils.py

```diff
@@ -0,0 +1,97 @@
+"""API utils."""
+
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def is_authenticated_path(event):
+    """Check if the API path is authenticated, allowing access to unreleased files.
+
+    This function examines the routeKey and rawPath in the event to determine
+    if the request is coming through an authenticated path (containing 'api-key'
+    or 'auth'). Authenticated paths have access to all files, while
+    non-authenticated paths only have access to released files.
+
+    Parameters
+    ----------
+    event : dict
+        The API Gateway event object
+
+    Returns
+    -------
+    bool
+        True if the path is authenticated, False otherwise
+    """
+    # Get the routeKey and rawPath, defaulting to empty strings if not present
+    auth_api_endpoint = ["api-key", "authorized"]
+
+    # Safely extract route_key and raw_path
+    route_key = ""
+    if event.get("routeKey") and "/" in event.get("routeKey"):
+        route_key = event.get("routeKey").split("/")[1]
```

Comment:
You can use the default return of get to shorten these up.
```suggestion
    rk = event.get("routeKey", "")
    route_key = rk.split("/")[1] if "/" in rk else ""
```

Reply:
that's what I had originally but AI changes

Reply:
See lower down for an even simpler update to everything to make this a one-line function.